src/common/paddle/organization.ts

```diff
@@ -0,0 +1,120 @@
+import type {
+  SubscriptionCreatedEvent,
+  SubscriptionCreatedNotification,
+} from '@paddle/paddle-node-sdk';
+import { randomUUID } from 'crypto';
+import createOrGetConnection from '../../db';
+import { extractSubscriptionCycle, paddleInstance } from '.';
+import { Organization, User } from '../../entity';
+import { ContentPreferenceOrganization } from '../../entity/contentPreference/ContentPreferenceOrganization';
+import { ContentPreferenceStatus } from '../../entity/contentPreference/types';
+import { OrganizationMemberRoles } from '../../roles';
+import { SubscriptionProvider, SubscriptionStatus } from '../plus';
+import { logger } from '../../logger';
+import { isPlusMember } from '../../paddle';
+import { updateSubscriptionFlags } from '../utils';
+
+type PaddleCustomData = {
+  user_id?: string;
+};
+
+// If the user provides business information during the checkout, we will use it
+// otherwise we will use the default business name
+const getBusinessName = async (
+  user: User,
+  data: SubscriptionCreatedNotification,
+) => {
+  const defaultBusinessName = `${user.name}'s new Organization`;
+  if (!data.businessId) {
+    return defaultBusinessName;
+  }
+
+  try {
+    const business = await paddleInstance.businesses.get(
+      data.customerId,
+      data.businessId,
+    );
+    return business.name;
+  } catch (error) {
+    return defaultBusinessName;
+  }
+};
+
+export const createOrganizationSubscription = async ({
+  event,
+}: {
+  event: SubscriptionCreatedEvent;
+}) => {
+  const organizationId = randomUUID();
+  const con = await createOrGetConnection();
+
+  const { data } = event;
+  const { user_id: userId }: PaddleCustomData = data?.customData || {};
+
+  const subscriptionType = extractSubscriptionCycle(data.items);
+  if (!subscriptionType) {
+    logger.error(
+      {
+        provider: SubscriptionProvider.Paddle,
```

Comment:
We might have to extend these a bit to ensure we quickly know what sub provider is failing?

Reply:
good idea. I'll add that after #2827 is merged and I merge into this branch